pkgs/development/libraries/mpich/default.nix

```diff
@@ -29,21 +42,38 @@ stdenv.mkDerivation  rec {
 
   outputs = [ "out" "doc" "man" ];
 
-  configureFlags = [
-    "--enable-shared"
-    "--with-pm=${withPmStr}"
-  ] ++ lib.optionals (lib.versionAtLeast gfortran.version "10") [
-    "FFLAGS=-fallow-argument-mismatch" # https://github.com/pmodels/mpich/issues/4300
-    "FCFLAGS=-fallow-argument-mismatch"
-  ] ++ lib.optionals pmixSupport [
-    "--with-pmix=${lib.getDev pmix}"
-  ];
+  configureFlags =
+    [
+      "--enable-shared"
+      "--with-pm=${withPmStr}"
+    ]
+    ++ lib.optionals (lib.versionAtLeast gfortran.version "10") [
+      "FFLAGS=-fallow-argument-mismatch" # https://github.com/pmodels/mpich/issues/4300
+      "FCFLAGS=-fallow-argument-mismatch"
+    ]
+    ++ lib.optionals withSlurm [
```

Comment:
This should be better called  `withPmi2Slurm`. Mpich also has a configure option `--with-slurm`, which probably does something else(?)

Reply:
Renamed to `withPmilib` accepting a string literal